runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1449,9 +1449,24 @@ TR_VectorAPIExpansion::isVectorizedOrScalarizedNode(TR::Node *node, TR::DataType
    int32_t refId = -1;
 
    if (node->getOpCodeValue() == TR::aload ||
-       node->getOpCodeValue() == TR::astore)
-      {
-      refId = node->getSymbolReference()->getReferenceNumber();
+       node->getOpCodeValue() == TR::astore ||
+       (node->getOpCode().isVectorOpCode() &&
+        (node->getOpCode().getVectorOperation() == TR::vload ||
+         node->getOpCode().getVectorOperation() == TR::vstore)
```

Comment:
Do we need to also check for scalar primitive loads and stores? Maybe not (yet?) based on the comment at the end of this if-else chain:
```
// TODO: check if node was already scalarized
```

Reply:
Yes, there might one place in the code where this method is called outside of boxing/unboxing. Otherwise, we currently don't allow scalarization with boxing/unboxing so it should work. But yes, strictly speaking we will need to check for that.

Reply:
OK. Since it's already not handled in general, I suppose it doesn't need to be part of this change